plugins/gatsby-theme-iterative-docs/src/utils/front/setUpCustomYtEmbeds.ts

```diff
@@ -0,0 +1,16 @@
+const setUpCustomYtEmbeds = () => {
```

Comment:
This script takes all of our youtube embeds and attaches a click listener to our overlay. If the overlay is clicked, the video is played and the overlay disappears. It's definitely not very "reacty"... Does anybody have any ideas on how we could improve this?

Once we agree on a implementation (whether we deside to do this one or a different way) I want to have the code update local storage so that the message won't appear again if the user revisits the browser :)

Reply:
> Does anybody have any ideas on how we could improve this?

Bit of a shot in the dark, but is it possible to have the custom embedder return a custom tag like `<youtube>` and then attach a React component to our `rehype-react` custom components with this logic in it? That way we can use the `remark-embedder` style of automatic URL grabbing with React JS more consistent with the rest of the site.

Reply:
I could try that but wouldn't that only work for our docs pages? We'd probably want a solution that works the same for blog and doc 🤔

Reply:
Good point, I forgot about our weird divergent blog markdown.